pkg/common/util.go

```diff
@@ -87,6 +87,40 @@ func ToLowerCaseKeys(keyValues map[string]string) map[string]string {
 	return newMap
 }
 
+// Recursively truncates the given "field" from CustomResourceDefinition to 50 characters.
+func truncateFieldRecursively(data map[string]interface{}, maxLength int, field string) {
+	for key, value := range data {
+		if key == field {
+			if str, ok := value.(string); ok && len(str) > maxLength {
+				data[key] = str[:maxLength]
+			}
+			continue
+		}
+		if subObj, ok := value.(map[string]interface{}); ok {
+			truncateFieldRecursively(subObj, 50, field)
```

Comment:
Why not pass `maxLength` here?
```suggestion
			truncateFieldRecursively(subObj, maxLength, field)
```